jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -418,11 +418,24 @@ public abstract boolean renameTo(Resource dest)
     /**
      * Returns the resource contained inside the current resource with the
      * given name, which may or may not exist.
+     * <p>
+     *     Constraints:
+     *     <ul>
+     *         <li>The path is expected to be in URI path-like format.</li>
+     *         <li>The path provided will not be URI decoded by the Resource implementation.</li>
```

Comment:
I think this is a bit of a convoluated way of saying that the path is decoded.  Ie any % characters will not result in further decoding, but will themselves be encoded